docs/src/index.md

```diff
@@ -1,12 +1,173 @@
-  of AlgebraicRewriting.
-- [Lotka Volterra example](https://github.com/AlgebraicJulia/AlgebraicRewriting.jl/blob/main/docs/src/lotka_volterra.jl): a model based 
-  on NetLogo's [wolf-sheep predation](http://ccl.northwestern.edu/netlogo/models/WolfSheepPredation) agent-based model demo.
-- [Game of Life example](https://github.com/AlgebraicJulia/AlgebraicRewriting.jl/blob/main/docs/src/GameOfLife.ipynb): an implementation of Conway's 
-  [game of life](https://en.wikipedia.org/wiki/Conway%27s_Game_of_Life) on an 
-  arbitrary graph (with the grid-structure as a special case).
\ No newline at end of file
+```@meta
+CurrentModule = AlgebraicRewriting
+```
+
+Algebraic rewriting is a context-aware find-and-replace operation that is useful for maintaining structure in various scenarios. This package provides tools for such operations in Julia, ensuring that rewrite rules adhere to structures defined using ACSets (see [ACSets.jl](https://github.com/AlgebraicJulia/ACSets.jl) and [Catlab.jl](https://github.com/AlgebraicJulia/Catlab.jl)).
+
+This page will provide you with a gentle overview of how to design and apply rewrite rules for a simple ACSet. **More sophisticated examples** can be found in the side-bar.
+
+## Setup Environment
+To begin, set up your environment by importing necessary packages.
+
+```julia
+using Catlab
+using AlgebraicRewriting
+```
+
+## Design a rewrite rule
+The general process for designing a rewrite rule is as follows:
+
+### 1. Define your schema 
+A schema defined by a finite presentation of a generalized algebraic theory model using generators, `Ob`, `Hom`, `AttrType`, and `Attr`.
+
+```julia
+@present SchTeam(FreeSchema) begin
+  Player::Ob
+  Team::Ob
+  IsMemberOf::Hom(Player, Team)
+
+  TeamName::AttrType
+  HasName::Attr(Team, TeamName)
+end
+```
+
+### 2. Create the schema type 
+Data for rules are stored in a data structure called an ACSet. 
+
+```julia
+@acset_type Team(SchTeam)
+```
+
+### 3. Define rule parts
+A rewrite rule consists of a span of ACSets (`L <-l- K -r-> R`), namely three ACSets (`L`, `K`, `R`) and two natural transformations (`l`, `r`):
+
+- Left ACSet, `L`, is the pre-condition for the rule to be applied.
+- Keep ACSet, `K`, is the data for the part of the state that remain consistent when the rule is applied.
+- Right ACSet, `R`, is the effect of the rule.
+- Left transformation, `l`, embeds `K` in `L`.
+- Right transformation, `r`, embed `K` in `R`.
+
+To define a rule, all five parts need to be defined. 
+
+It is possible to insert data according to the schema using a **static** approach _or_ the **colimit-of-representables** approach.
+
+#### Static Instantiation (`@acset`)
+If using the **static** approach, you must fully specify the ACSet functors and natural transformation. Here is a rule that defines the ACSet statically. 
+
+In this example, the rule swaps player between two teams. 
```

Comment:
swaps player**s**?

Also the rule doesn't swap player #1 and #4, not sure if that's intended.

Reply:
Yeah, that is. It was intended to simulate a trade of players, one from each team. I revised the sentence to hopefully make it clearer.